src/VectorBundle.jl

```diff
@@ -0,0 +1,372 @@
+    error("sharp_isomorphism! not implemented for vector bundle fibers space " *
+        "of type $(typeof(M)), vector of type $(typeof(v)), point of " *
+        "type $(typeof(x)) and vector of type $(typeof(w)).")
+end
+
+function sharp_isomorphism(M::Manifold, x, w::FVector)
+    v = similar_result(M, sharp_isomorphism, w, x)
+    sharp_isomorphism!(M, v, x, w)
+    return v
+end
+
+function similar_result(M::Manifold, ::typeof(sharp_isomorphism), w::FVector{CotangentSpaceType}, x)
+    return FVector(TangentSpace, similar(w.data))
+end
+
+"""
+    similar_result_type(M::VectorBundleFibers, f, args::NTuple{N,Any}) where N
+
+Returns type of element of the array that will represent the result of
+function `f` for representing an operation with result in the vector space `VS`
+for manifold `M` on given arguments (passed at a tuple).
+"""
+function similar_result_type(M::VectorBundleFibers, f, args::NTuple{N,Any}) where N
+    T = typeof(reduce(+, one(eltype(eti)) for eti ∈ args))
+    return T
+end
+
+"""
+    similar_result(M::VectorBundleFibers, f, x...)
+
+Allocates an array for the result of function `f` that is
+an element of the vector space of type `M.VS` on manifold `M.M`
+and arguments `x...` for implementing the non-modifying operation
+using the modifying operation.
+"""
+function similar_result(M::VectorBundleFibers, f, x...)
+    T = similar_result_type(M, f, x)
+    return similar(x[1], T)
+end
+
+norm(M::VectorBundleFibers, x, v) = sqrt(inner(M, x, v, v))
+
+norm(M::VectorBundleFibers{<:TangentSpaceType}, x, v) = norm(M.M, x, v)
+
+"""
+    vector_distance(M::VectorBundleFibers, x, v, w)
+
+Distance between vectors `v` and `w` from the vector space at point `x`
+from the manifold `M.M`, that is the base manifold of `M`.
+"""
+vector_distance(M::VectorBundleFibers, x, v, w) = norm(M, x, v-w)
+
+"""
+    inner(M::VectorBundleFibers, x, v, w)
+
+Inner product of vectors `v` and `w` from the vector space of type `VS`
+at point `x` from manifold `M`.
+"""
+function inner(M::VectorBundleFibers, x, v, w)
+    error("inner not defined for vector space family of type $(typeof(M)), " *
+        "point of type $(typeof(x)) and " *
+        "vectors of types $(typeof(v)) and $(typeof(w)).")
+end
+
+function inner(M::VectorBundleFibers{<:TangentSpaceType}, x, v, w)
+    return inner(M.M, x, v, w)
+end
+
+function inner(M::VectorBundleFibers{<:CotangentSpaceType}, x, v, w)
+    return inner(M.M, x, flat_isomorphism(M, x, v), flat_isomorphism(M, x, w))
+end
+
+"""
+    vector_space_dimension(M::VectorBundleFibers)
+
+Dimension of the vector space of type `M`.
+"""
+function vector_space_dimension(M::VectorBundleFibers)
+    error("vector_space_dimension not implemented for vector space family $(typeof(M)).")
+end
+
+vector_space_dimension(M::VectorBundleFibers{<:TCoTSpaceType}) = manifold_dimension(M.M)
+
+function vector_space_dimension(M::VectorBundleFibers{<:TensorProductType})
```

Comment:
I haven't tested this, but I think it can be implemented more efficiently with `mapreduce`?

Reply:
I don't think it makes much of a difference anyway. Allocating objects of type `VectorBundleFibers` is what takes the most time.